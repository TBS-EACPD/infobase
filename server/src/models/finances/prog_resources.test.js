const query = `
query ($program_id: String, $dept_code: String){
  root(lang: "en"){
    program(id: $program_id){
      program_spending_data {
        ...prog_spend_test
      }
      program_fte_data {
        ...prog_fte_test
      }
    }
    org(dept_code: $dept_code){
      program_spending_data {
        ...prog_spend_test
      }
      program_fte_data {
        ...prog_fte_test
      }
    }
  }
}


fragment prog_spend_test on ProgramSpendingData {

  totals {
    amount
    year
  }

  basic_spend_trend {
    pa_last_year_exp
    pa_last_year_auth
    pa_last_year_planned
    planning_year_1
    historical_change_pct
    avg_all
    avg_planned
    avg_historical
  }

}

fragment prog_fte_test on ProgramFteData {


  totals {
    amount
    year
  }

  basic_fte_trend{
    pa_last_year
    historical_change_pct
    planned_change_pct
    avg_all
    avg_planned
    avg_historical
  }
}`


const { execQuery } = global;

describe("program resources data", function(){

  it("program resource snapshot, org (AGR) and prog (AGR-AAA00)", async ()=> {
    const variables = { 
      dept_code: "AGR",
      program_id: "AGR-AAA00",
    };
    const data = await execQuery(query, variables);
    return expect(data).toMatchSnapshot();
  });

})