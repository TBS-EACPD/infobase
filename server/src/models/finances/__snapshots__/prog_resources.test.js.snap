// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`program resources data program resource snapshot, org (AGR) and prog (AGR-AAA00) 1`] = `
Object {
  "data": Object {
    "root": Object {
      "org": Object {
        "program_fte_data": Object {
          "basic_fte_trend": Object {
            "avg_all": 4788.125,
            "avg_historical": 4888.4,
            "avg_planned": 4621,
            "historical_change_pct": -0.18862592723419286,
            "pa_last_year": 4594,
            "planned_change_pct": 0.00239442751414889,
          },
          "totals": Array [
            Object {
              "amount": 5662,
              "year": "pa_last_year_5",
            },
            Object {
              "amount": 5076,
              "year": "pa_last_year_4",
            },
            Object {
              "amount": 4597,
              "year": "pa_last_year_3",
            },
            Object {
              "amount": 4513,
              "year": "pa_last_year_2",
            },
            Object {
              "amount": 4594,
              "year": "pa_last_year",
            },
            Object {
              "amount": 4653,
              "year": "planning_year_1",
            },
            Object {
              "amount": 4605,
              "year": "planning_year_2",
            },
            Object {
              "amount": 4605,
              "year": "planning_year_3",
            },
          ],
        },
        "program_spending_data": Object {
          "basic_spend_trend": Object {
            "avg_all": 2016745800.75,
            "avg_historical": 2325153743,
            "avg_planned": 1502732563.6666667,
            "historical_change_pct": -0.02485446829098852,
            "pa_last_year_auth": 2703946959,
            "pa_last_year_exp": 2614910350,
            "pa_last_year_planned": 2263733256,
            "planning_year_1": 2251183698,
          },
          "totals": Array [
            Object {
              "amount": 2884155827,
              "year": "pa_last_year_5_auth",
            },
            Object {
              "amount": 2681559075,
              "year": "pa_last_year_5_exp",
            },
            Object {
              "amount": 2556585385,
              "year": "pa_last_year_4_auth",
            },
            Object {
              "amount": 2386898330,
              "year": "pa_last_year_4_exp",
            },
            Object {
              "amount": 2113152417,
              "year": "pa_last_year_3_auth",
            },
            Object {
              "amount": 2013991368,
              "year": "pa_last_year_3_exp",
            },
            Object {
              "amount": 2040884675,
              "year": "pa_last_year_2_auth",
            },
            Object {
              "amount": 1928409592,
              "year": "pa_last_year_2_exp",
            },
            Object {
              "amount": 2703946959,
              "year": "pa_last_year_auth",
            },
            Object {
              "amount": 2614910350,
              "year": "pa_last_year_exp",
            },
            Object {
              "amount": 2263733256,
              "year": "pa_last_year_planned",
            },
            Object {
              "amount": 2251183698,
              "year": "planning_year_1",
            },
            Object {
              "amount": 1130438030,
              "year": "planning_year_2",
            },
            Object {
              "amount": 1126575963,
              "year": "planning_year_3",
            },
          ],
        },
      },
      "program": Object {
        "program_fte_data": Object {
          "basic_fte_trend": Object {
            "avg_all": 365,
            "avg_historical": 364.4,
            "avg_planned": 366,
            "historical_change_pct": -0.021333333333333333,
            "pa_last_year": 367,
            "planned_change_pct": -0.0027247956403269754,
          },
          "totals": Array [
            Object {
              "amount": 375,
              "year": "pa_last_year_5",
            },
            Object {
              "amount": 370,
              "year": "pa_last_year_4",
            },
            Object {
              "amount": 349,
              "year": "pa_last_year_3",
            },
            Object {
              "amount": 361,
              "year": "pa_last_year_2",
            },
            Object {
              "amount": 367,
              "year": "pa_last_year",
            },
            Object {
              "amount": 366,
              "year": "planning_year_1",
            },
            Object {
              "amount": 366,
              "year": "planning_year_2",
            },
            Object {
              "amount": 366,
              "year": "planning_year_3",
            },
          ],
        },
        "program_spending_data": Object {
          "basic_spend_trend": Object {
            "avg_all": 1071225020.125,
            "avg_historical": 1186937096.4,
            "avg_planned": 878371559.6666666,
            "historical_change_pct": -0.10076136456697256,
            "pa_last_year_auth": 1295672303,
            "pa_last_year_exp": 1276920061,
            "pa_last_year_planned": 1305927027,
            "planning_year_1": 1306311287,
          },
          "totals": Array [
            Object {
              "amount": 1434879185,
              "year": "pa_last_year_5_auth",
            },
            Object {
              "amount": 1420001333,
              "year": "pa_last_year_5_exp",
            },
            Object {
              "amount": 1295485113,
              "year": "pa_last_year_4_auth",
            },
            Object {
              "amount": 1280762636,
              "year": "pa_last_year_4_exp",
            },
            Object {
              "amount": 1040190844,
              "year": "pa_last_year_3_auth",
            },
            Object {
              "amount": 1033315991,
              "year": "pa_last_year_3_exp",
            },
            Object {
              "amount": 944761735,
              "year": "pa_last_year_2_auth",
            },
            Object {
              "amount": 923685461,
              "year": "pa_last_year_2_exp",
            },
            Object {
              "amount": 1295672303,
              "year": "pa_last_year_auth",
            },
            Object {
              "amount": 1276920061,
              "year": "pa_last_year_exp",
            },
            Object {
              "amount": 1305927027,
              "year": "pa_last_year_planned",
            },
            Object {
              "amount": 1306311287,
              "year": "planning_year_1",
            },
            Object {
              "amount": 664401696,
              "year": "planning_year_2",
            },
            Object {
              "amount": 664401696,
              "year": "planning_year_3",
            },
          ],
        },
      },
    },
  },
}
`;
